planning/behavior_path_planner/src/utils/utils.cpp

```diff
@@ -621,6 +621,29 @@ bool checkCollisionBetweenPathFootprintsAndObjects(
   return false;
 }
 
+bool checkCollisionWithExtraStoppingMargin(
+  const PathWithLaneId & ego_path, const PredictedObjects & dynamic_objects,
+  const double base_to_front, const double base_to_rear, const double width,
+  const double maximum_deceleration, const double margin, const double max_extra_stopping_margin)
```

Comment:
how about 
`collision_distance_threshold` or `collision_distance` hmm.. something better variable name for `margin`?

Reply:
rename to collision_check_margin considering params

https://github.com/autowarefoundation/autoware.universe/pull/4674/commits/99229aac24bc6c2550fc888f153dca6883762947